kedro-datasets/kedro_datasets/api/api_dataset.py

```diff
@@ -76,6 +74,8 @@ def __init__(
             credentials: Allows specifying secrets in credentials.yml.
                 Expected format is ``('login', 'password')`` if given as a tuple or list.
                 An ``AuthBase`` instance can be provided for more complex cases.
+            metadata: Any arbitrary user metadata.
```

Comment:
I think it would be good to mention here that Kedro doesn't do anything with this metadata, but that it can be consumed by plugins or directly by the user.

Reply:
> > How was `layer` being used before without being an argument in `DataSet`? I think it should be yes.
> 
> `layer` is currently consumed by the DataCatalog and the general consensus is that this is not ideal. Previously it was [defined on every dataset](https://github.com/quantumblacklabs/private-kedro/pull/548/files) as in this PR
> 
> > It would be also great to think about how metadata could be access via hooks, i.e. how will viz using this new field. after_caterlog_create hook I guess?
> 
> `metadata` is accessible in the dummy hook implementation that I used to test the datasets (in PR description) , for the specifics on how viz (and other plugins) would consume and use this however, I am not sure

Hey @AhdraMeraliQB  -- i think this is very important because currently we get layer information when we context.catalog -- would the metadata information also be in that way?

Reply:
We would need to change how it's consumed on kedro-viz so that instead of looking at `catalog.layers` we look at `get_dataset(dataset_name).metadata["kedro-viz"]["layer"]` (where `get_dataset` is defined in `CatalogRepository`).